lib/msal-common/src/network/NetworkManager.ts

```diff
@@ -43,23 +43,62 @@ export class NetworkManager {
     }
 
     /**
-     * Wraps sendPostRequestAsync with necessary preflight and postflight logic
+     * Wraps the networkClient's sendGetRequestAsync with necessary preflight and postflight logic
+     * Managed Identity sends GET requests to communicate to the following managed identity sources:
+     * App Service, Azure Arc, Imds, Service Fabric
+     *
+     * @param thumbprint
+     * @param tokenEndpoint
+     * @param options
+     */
+    async sendGetRequest<T extends ServerAuthorizationTokenResponse>(
```

Comment:
We discussed that `NetworkManager` is actually `TokenEndpointNetworkManager`. But in MSI case, you'll be hitting different endpoints, which will not return `ServerAuthorizationTokenResponse`. So I don't think `NetworkManager` can be used here. 

Consider creating an `MsiNetworkManager` or using the `networkClient` directly.

Reply:
Based on your other comment about MSI not needing throttling, I think it makes more sense to just call the networkClient directly.